### PR TITLE
Rename-DbaDatabase, round #3

### PR DESCRIPTION
**Pull request by niphlod**
## Type of Change

 - [ ] Bug fix (non-breaking change, fixes #<enter issue number>)
 - [x] New feature (non-breaking change, adds functionality)
 - [ ] Breaking change (effects multiple commands or functionality)
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 
### Purpose
Reissue of  https://github.com/sqlcollaborative/dbatools/pull/2442, which is a reissue of https://github.com/sqlcollaborative/dbatools/pull/2030 . Please merge ASAP, keeping tracks is starting to be extremely difficult.

### CODE REVIEW

File: functions/Rename-DbaDatabase.ps1 [hunk 1]
```diff
@@ -0,0 +1,870 @@
...
+		By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+		This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+		Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+	.NOTES
+		Author: niphlod
+
+		dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+		Copyright (C) 2016 Chrissy LeMaire
+		This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
```
Comment (wsmelton) [inline functions/Rename-DbaDatabase.ps1]: Need to update this to current license statement

**Comment by wsmelton** [comment]
Just curious if the `-Preview` output should look like this...It didn't actual do any changes, but the way that output appears it looks like it did unless you pay close attention to the output:
image

Shouldn't that be using `WhatIf` instead of having a parameter for it called `-Preview`?

Just one more small thing...if an operation requires setting the database offline and that fails I don't think it is best to keep going with the changes and leaving "low hanging fruit" like "You are in charge of moving the files to the new location".
image

I obviously don't know how to use this one 😝 
image

I think overall it is a good first round but believe it is going to be too easy for folks to dork up their databases with those placeholders. We will likely need a good blog post or something to point folks at starting out. Then as we get more folks using it figure out how (or if we can) fine tune it.

**Comment by wsmelton** [comment]
I'm up for merging it but will leave that to @potatoqualitee ... I mean we have to start somewhere and just wait for feedback to come in on it.

**Comment by niphlod** [comment]
yeah, apparently no better ideas came around during code review so we should hear from users pretty soon. The original requested feature-set is in. Lemme fix the license and try to figure out your issue, then I think it can be merged.

**Comment by potatoqualitee** [comment]
I'd love to merge! Then we'll ask for some testers.  @wsmelton - feel free to merge and ping me once you and niph are ready. then i'll add it to psd1

**Comment by wsmelton** [comment]
I would say we push this one out as a "beta" type thing, get folks using it in their test environments and then have them offer feedback/suggestions. Especially since we cannot test for every possible scenario of file configurations in a database.

**Comment by niphlod** [comment]
on the "leave the low hanging fruit" theme, unfortunately it's kinda chicken/egg problem: nothing can be moved until the db is put offline (files are locked if db is online), and there's no way to pre-check if a move can be done. Maybe it should try to rollback the changes, but it's not that simple.

@wsmelton : the error shown seems to point out you can't `Get-DbaFile -sqlinstance matanarms\sql11 -path 'c:\program files\microsoft sql server\mssql11.sql11\mssql\data'` , can you confirm ?

**Comment by potatoqualitee** [comment]
Yesssss thanks @niphlod ! Adding this and alerting Rob Volk who originally requested it.

**Pull request merged**